terraform/modules/tf_globals/alerts_firehose/main.tf

```diff
@@ -1,28 +1,80 @@
 locals {
-  stream_name = "${var.prefix}_streamalert_alert_delivery"
+  stream_name         = "${var.prefix}_streamalert_alert_delivery"
+  bucket_arn          = "arn:aws:s3:::${var.prefix}-streamalerts"
+  alerts_location     = "s3://${var.prefix}-streamalerts/alerts"
+  ser_de_params_key   = var.store_format == "parquet" ? "serialization.format" : "ignore.malformed.json"
+  ser_de_params_value = var.store_format == "parquet" ? "1" : "true"
 }
 
-// AWS Firehose Stream for Alerts to S3
 resource "aws_kinesis_firehose_delivery_stream" "streamalerts" {
   name        = local.stream_name
-  destination = "s3"
-
-  s3_configuration {
-    role_arn           = aws_iam_role.firehose.arn
-    bucket_arn         = "arn:aws:s3:::${var.bucket_name}"
-    prefix             = "alerts/"
-    buffer_size        = var.buffer_size
-    buffer_interval    = var.buffer_interval
-    compression_format = var.compression_format
-    kms_key_arn        = var.kms_key_arn
-
-    cloudwatch_logging_options {
-      enabled         = true
-      log_group_name  = aws_cloudwatch_log_group.firehose.name
-      log_stream_name = "S3Delivery"
+  destination = var.store_format == "parquet" ? "extended_s3" : "s3"
+
+  // AWS Firehose Stream for Alerts to S3 and saved in JSON format
+  dynamic "s3_configuration" {
+    for_each = var.store_format == "parquet" ? [] : [var.store_format]
+    content {
+      role_arn           = aws_iam_role.firehose.arn
+      bucket_arn         = local.bucket_arn
+      prefix             = "alerts/"
+      buffer_size        = var.buffer_size
+      buffer_interval    = var.buffer_interval
+      compression_format = "GZIP"
+      kms_key_arn        = var.kms_key_arn
+
+      cloudwatch_logging_options {
+        enabled         = true
+        log_group_name  = aws_cloudwatch_log_group.firehose.name
+        log_stream_name = "S3Delivery"
+      }
+    }
+  }
+
+  // AWS Firehose Stream for Alerts to S3 and saved in Parquet format
+  dynamic "extended_s3_configuration" {
```

Comment:
I ran into this problem during Normalization v2. Be very careful because changing from `s3` —> `extended_s3` will destroy your existing Firehose. Because Firehose is stateful it can cause a small amount of data loss. Just FYI to consider for the migration process

Reply:
Good catch!!!